arangod/IResearch/IResearchDataStore.cpp

```diff
@@ -568,6 +572,14 @@ IResearchDataStore::Snapshot IResearchDataStore::snapshot() const {
         << id() << "'";
     return {};  // return an empty reader
   }
+  if (failQueriesOnOutOfSync() && linkLock->isOutOfSync()) {
+    // link has failed, we cannot use it for querying
+    THROW_ARANGO_EXCEPTION_MESSAGE(
+        TRI_ERROR_CLUSTER_AQL_COLLECTION_OUT_OF_SYNC,
+        absl::StrCat("link ", std::to_string(linkLock->id().id()),
+                     " has been marked as failed and needs to be recreated"));
```

Comment:
maybe `is out of sync and needs to be recreated`?

Reply:
done